forest/benchmarking/observable_estimation.py

```diff
@@ -1151,3 +1150,70 @@ def ratio_variance(a: Union[float, np.ndarray],
     :param var_b: Variance in 'B'
     """
     return var_a / b**2 + (a**2 * var_b) / b**4
+
+
+def merge_disjoint_experiments(experiments: List[ObservablesExperiment]) -> ObservablesExperiment:
+    """
+    Merges the list of experiments into a single experiment that runs the sum of the individual
+    experiment programs and contains all of the combined experiment settings.
+
+    A group of ObservablesExperiments whose programs operate on disjoint sets of qubits can be
+    'parallelized' so that the total number of runs can be reduced after grouping the settings.
+    Settings which act on disjoint sets of qubits can be automatically estimated from the same
+    run on the quantum computer.
+
+    If any experiment programs act on a shared qubit they cannot be thoughtlessly composed since
+    the order of operations on the shared qubit may have a significant impact on the program
+    behaviour; therefore we do not recommend using this method if this is the case.
+
+    Even when the individual experiments act on disjoint sets of qubits you must be
+    careful not to associate 'parallel' with 'simultaneous' execution. Physically the gates
+    specified in a pyquil Program occur as soon as resources are available; meanwhile, measurement
+    happens only after all gates. There is no specification of the exact timing of gates beyond
+    their causal relationships. Therefore, while grouping experiments into parallel operation can
+    be quite beneficial for time savings, do not depend on any simultaneous execution of gates on
+    different qubits, and be wary of the fact that measurement happens only after all gates have
+    finished.
+
+    Note that to get the time saving benefits you need to call `group_settings` on the ouput
+    experiment.
```

Comment:
Is there are reason why we don't do this grouping for the user already? Or have it as an option?

Reply:
There are different ways of grouping so I didn't want to force one. But I think your suggestion for the option is prudent since I imagine it will be the common use case.